Code/A06/devterm-a06-gearbox

```diff
@@ -166,91 +215,62 @@ class A06:
-            self.set_cpu_on_off(2,1)
-            self.set_cpu_on_off(3,1)
-            self.set_cpu_on_off(4,0)
-            self.set_cpu_on_off(5,0)
-            self.set_cpu_max_freq(0,1008000)
-            self.set_cpu_max_freq(1,1008000)
-            self.set_cpu_max_freq(2,1008000)
-            self.set_cpu_max_freq(3,1008000)
-            self.set_gpu(self.gpu_governor,self.gpu_max_freq*2)
+        self.gear = load_gear(g)
+
+        for (cpu, freq) in enumerate(self.gear["cpu"]):
+            enabled = freq > 0
+            self.set_cpu_on_off(cpu, int(enabled))
+            if enabled:
+                self.set_cpu_max_freq(cpu, freq)
+
+        self.set_gpu(self.gear["gpu_gov"], self.gear["gpu_freq"])
+
+        # TODO: Generalise this
+        if self.gear["cpu"][0] > 0:
             self.set_cpu_gov0(self.cpu_scaling_governor)
-        if self.gear == 4:
-            self.set_cpu_on_off(4,1)
-            self.set_cpu_on_off(5,1)
-            self.set_cpu_on_off(0,0)
-            self.set_cpu_on_off(1,0)
-            self.set_cpu_on_off(2,0)
-            self.set_cpu_on_off(3,0)
-            self.set_cpu_max_freq(4,1008000)
-            self.set_cpu_max_freq(5,1008000)
-            self.set_gpu(self.gpu_governor,self.gpu_max_freq*2)
-            self.set_cpu_gov4(self.cpu_scaling_governor)
-        if self.gear == 5:
-            self.set_cpu_on_off(4,1)
-            self.set_cpu_on_off(5,1)
-            self.set_cpu_on_off(0,0)
-            self.set_cpu_on_off(1,0)
-            self.set_cpu_on_off(2,0)
-            self.set_cpu_on_off(3,0)
-            self.set_cpu_max_freq(4,1200000)
-            self.set_cpu_max_freq(5,1200000)
-            self.set_gpu(self.gpu_governor,self.gpu_max_freq*2)
+        else:
             self.set_cpu_gov4(self.cpu_scaling_governor)
 
 
 
 def print_gear_map(gear):
-    global gear_m
-    rp = -1
+    print("    +-----------------------------------+-----------------+-----------+")
+    print("    |            Cortex-A53             |   Cortex-A72    | Mali-T860 |")
+    print("    +--------+--------+--------+--------+--------+--------+-----------+")
+    print("    | CPU 0  | CPU 1  | CPU 2  | CPU 3  | CPU 4  | CPU 5  |   GPU     |")
+    div = "+---+--------+--------+--------+--------+--------+--------+-----------+"
+    print(div)
 
-    if gear > 0:
-        rp = 5 + (gear -1)*2
-
-    for idx, val in enumerate(gear_m):
-        if idx == rp:
-            print(gear_idx[gear-1])
+    def freq(khz):
+        mhz = khz/1000
+        if mhz >= 1000:
+            return "%d MHz" % mhz
+        elif mhz > 0:
+            return " %d MHz" % mhz
         else:
-            print(val) 
-      
+            return "  OFF   "
+
+    for idx, val in enumerate(gears):
+        g = idx + 1
+        selected = g == gear
+        print("|%s|%s| %s  |%s" % (
+            ("*%s*" if selected else " %s ") % g,
+            "|".join([freq(cpu) for cpu in val["cpu"]]),
+            freq(val["gpu_freq"]/1000),
```

Comment:
This looks a little less pretty than before, but I decided it wasn't worth the extra code to generate the pretty format from the table of gears.